### ISSUE TITLE
Prevent sensors from getting stale.

### ISSUE THREAD

**Issue by amlinger** [amlinger/homebridge-telldus-tdtool#30]
The original source of this problem was first addressed in #29, as `tdtool` spawns new sensors if the connection is poor. A solution could be using the config to select which sensors to use.

**Comment by jannylund** [comment]
Here's a suggested configuration for sensors. Some reasoning:
1. If `config["sensors"]` exist, we should not autoconfigure any sensors from `tdtool --list-sensors` because;
   1.1 this allows us to avoid any false sensors.
   1.2 this allows us to register sensors even after telldusd has just been restarted and/or is down.
   1.3. this allows us to register sensors that are currently out of battery.
2. id is required to identify the sensors.
3. model is required to be able to autoregister sensors on startup even if they are not visible in tdtool at that point in time.
4. model can be used to override the type that comes from tdtool (removing the need of one PR I made earlier with sensors identifying them self wrongly).
5. name can be used instead of the autogenerated name.

```
"platforms": [
        {
            "name": "TellstickDuo",
            "platform": "Telldus-TD-Tool",
            "sensors": [
                {
                    "id": 114,
                    "model": "temperaturehumidity",
                    "name": "Bathroom"
                },
                {
                    "id": 142,
                    "model": "temperature",
                    "name": "Sauna"
                },
                {
                    "id": 74,
                    "model": "temperature",
                    "name": "Freezer"
                }
            ]
        }
    ]
```

**Comment by jannylund** [comment]
Potentially we could add an `maxAge` parameter on each sensor, or on some higher level. But the problem is still what we should return to make homebridge happy if the sensor is not responding.

**Comment by amlinger** [comment]
I very much agree with point 1, nothing should be different if there is no config. I'd very much keep away from relying on id's, though, if we can think of a solution. The problem is that as soon as a sensor gets stale, or as soon as `telldusd` is restarted somehow, we need to reconfigure this, which seems very tedious to me (as far as I've seen, the id of tdtool sensors cannot really be trusted). 
I strongly like the benefit of your point 3 as well, as there are no race conditions between `telldusd` and Homebridge. 
Supplying quantities of expected sensors would probably solve them going stale, but will make reliable names unlikely, and naming on startup in the config impossible.

**Comment by jannylund** [comment]
Since there is no hw id available in the protocol used by these sensors, id is the only thing we can rely on. Id is also fixed, the only time it can change is when batteries are exchanged in the sensors (at least the ESIC sensors reset id if you pull the batteries). This should not be a problem though, as predefining the id's in config means they sensors will not show up until they are configured to use the proper id.

**Comment by amlinger** [comment]
I see! Then I definitely agree, as long as we can rely on the ID for the remaining battery-life, I believe it is fine. I believe that the solution that you are proposing is the one we should implement.

**Comment by jannylund** [comment]
Configuration part is working in https://github.com/amlinger/homebridge-telldus-tdtool/pull/31

The stale part is still to be solved. Now it just shows old information which is not very nice...

**Comment by jannylund** [comment]
And the timeout part is fixed in https://github.com/amlinger/homebridge-telldus-tdtool/pull/32

**Comment by amlinger** [comment]
Thanks a lot for taking the time! Sorry that it's taken a while to get to this, I've been moving all weekend and haven't had a chance to yet. I'll look it over tonight.

**Comment by amlinger** [comment]
Should be closed by #31 and #32.
